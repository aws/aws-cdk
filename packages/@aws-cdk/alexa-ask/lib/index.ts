// Alexa::ASK CloudFormation Resources:
export * from './ask.generated';
export * from './pipeline-actions';
