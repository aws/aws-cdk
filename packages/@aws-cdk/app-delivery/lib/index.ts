export * from './pipeline-deploy-stack-action';
