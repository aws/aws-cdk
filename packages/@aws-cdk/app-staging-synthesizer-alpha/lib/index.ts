export * from './default-staging-stack';
export * from './app-staging-synthesizer';
export * from './bootstrap-roles';
export * from './staging-stack';
