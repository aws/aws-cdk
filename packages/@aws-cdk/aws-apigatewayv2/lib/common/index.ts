export * from './integration';
export * from './route';
export * from './stage';
export * from './domain-name';
export * from './api-mapping';
