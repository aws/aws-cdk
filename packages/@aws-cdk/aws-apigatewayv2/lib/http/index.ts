export * from './api';
export * from './route';
export * from './integration';
export * from './integrations';
export * from './stage';