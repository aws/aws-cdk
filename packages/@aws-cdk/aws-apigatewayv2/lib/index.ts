export * from './apigatewayv2.generated';
export * from './common';
export * from './http';