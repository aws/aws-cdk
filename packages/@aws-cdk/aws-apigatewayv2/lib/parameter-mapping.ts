/**
 * Represents a Mapping Value.
 */
export interface IMappingValue {
  /**
   * Represents a Mapping Value.
   */
  readonly value: string;
};

/**
 * Represents a Mapping Value.
 */
export class MappingValue implements IMappingValue {
  /**
  * Creates an empty mapping value.
  */
  public static readonly NONE = new MappingValue('');

  /**
   * Creates a header mapping value.
   */
  public static requestHeader(name: string) { return new MappingValue(`$request.header.${name}`); }

  /**
   * Creates a query string mapping value.
   */
  public static requestQueryString(name: string) { return new MappingValue(`$request.querystring.${name}`); }

  /**
   * Creates a request body mapping value.
   */
  public static requestBody(name: string) { return new MappingValue(`$request.body.${name}`); }

  /**
   * Creates a request path mapping value.
   */
  public static requestPath() { return new MappingValue('$request.path'); }

  /**
   * Creates a request path parameter mapping value.
   */
  public static requestPathParam(name: string) { return new MappingValue(`$request.path.${name}`); }

  /**
   * Creates a context variable mapping value.
   */
  public static contextVariable(variableName: string) { return new MappingValue(`$context.${variableName}`); }

  /**
   * Creates a stage variable mapping value.
   */
  public static stageVariable(variableName: string) { return new MappingValue(`$stageVariables.${variableName}`); }

  /**
   * Creates a custom mapping value.
   */
  public static custom(value: string) { return new MappingValue(value); }

  /**
   * Represents a Mapping Value.
   */
  public readonly value: string

  protected constructor(value: string) {
    this.value = value;
  }
}

/**
 * Represents a Parameter Mapping.
 */
export class ParameterMapping {

  /**
   * Creates a mapping from an object.
   */
  public static fromObject(obj: { [key: string]: MappingValue }): ParameterMapping {
    const mapping = new ParameterMapping();
    for (const [k, m] of Object.entries(obj)) {
      mapping.custom(k, m.value);
    }
    return mapping;
  }

  /**
  * Represents all created parameter mappings.
  */
  public readonly mappings: { [key: string]: string }
  constructor() {
    this.mappings = {};
  }

  /**
   * Creates a mapping to append a header.
   */
  public appendHeader(name: string, value: MappingValue): ParameterMapping {
    this.mappings[`append:header.${name}`] = value.value;
    return this;
  }

  /**
   * Creates a mapping to overwrite a header.
   */
  public overwriteHeader(name: string, value: MappingValue): ParameterMapping {
    this.mappings[`overwrite:header.${name}`] = value.value;
    return this;
  }

  /**
   * Creates a mapping to remove a header.
   */
  public removeHeader(name: string): ParameterMapping {
    this.mappings[`remove:header.${name}`] = '';
    return this;
  }

  /**
   * Creates a mapping to append a query string.
   */
  public appendQueryString(name: string, value: MappingValue): ParameterMapping {
    this.mappings[`append:querystring.${name}`] = value.value;
    return this;
  }

  /**
   * Creates a mapping to overwrite a querystring.
   */
  public overwriteQueryString(name: string, value: MappingValue): ParameterMapping {
    this.mappings[`overwrite:querystring.${name}`] = value.value;
    return this;
  }

  /**
   * Creates a mapping to remove a querystring.
   */
  public removeQueryString(name: string): ParameterMapping {
    this.mappings[`remove:querystring.${name}`] = '';
    return this;
  }

  /**
   * Creates a mapping to overwrite a path.
   */
  public overwritePath(value: MappingValue): ParameterMapping {
    this.mappings['overwrite:path'] = value.value;
    return this;
  }

  /**
   * Creates a custom mapping.
   */
  public custom(key: string, value: string): ParameterMapping {
    this.mappings[key] = value;
    return this;
  }
}
