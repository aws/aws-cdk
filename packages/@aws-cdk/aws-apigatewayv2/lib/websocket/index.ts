export * from './api';
export * from './route';
export * from './stage';
export * from './integration';
export * from './authorizer';
