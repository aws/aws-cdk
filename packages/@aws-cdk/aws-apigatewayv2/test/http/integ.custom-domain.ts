import * as acm from '@aws-cdk/aws-certificatemanager';
import * as lambda from '@aws-cdk/aws-lambda';
import { App, CfnOutput, Stack } from '@aws-cdk/core';
import { DomainName, HttpApi, LambdaProxyIntegration } from '../../lib';

const app = new App();

const stack = new Stack(app, 'integ-http-proxy');

const handler = new lambda.Function(stack, 'echohandler', {
  runtime: lambda.Runtime.NODEJS_12_X,
  handler: 'index.handler',
  code: new lambda.InlineCode('exports.handler = async function(event, context) { return { statusCode: 200, headers: { "content-type": "application/json"  }, body: JSON.stringify(event) }; };'),
});

const certArn = 'arn:aws:acm:us-east-1:111111111111:certificate';
const domainName = 'apigv2.demo.com';

const dn = new DomainName(stack, 'DN', {
  domainName,
  certificate: acm.Certificate.fromCertificateArn(stack, 'cert', certArn),
});

const prodApi = new HttpApi(stack, 'HttpProxyProdApi', {
  defaultIntegration: new LambdaProxyIntegration({ handler }),
  // https://${dn.domainName} goes to prodApi $default stage
  defaultDomainMapping: {
    domainName: dn,
    mappingKey: '/',
  },
});

const betaApi = new HttpApi(stack, 'HttpProxyBetaApi', {
  defaultIntegration: new LambdaProxyIntegration({ handler }),
  // https://${dn.domainName}/beta goes to betaApi $default stage
  defaultDomainMapping: {
    domainName: dn,
    mappingKey: 'beta',
  },
});

prodApi.addStage('testing', {
  stageName: 'testing',
  autoDeploy: true,
  // https://${dn.domainName}/testing goes to prodApi testing stage
  domainMapping: {
    domainName: dn,
    mappingKey: 'testing',
  },
} );

new CfnOutput(stack, 'RegionalDomainName', { value: dn.regionalDomainName });
new CfnOutput(stack, 'DomainName', { value: dn.domainName });
new CfnOutput(stack, 'CustomUDomainURL', { value: `https://${dn.domainName}` });
new CfnOutput(stack, 'ProdApiEndpoint', { value: prodApi.url! });
new CfnOutput(stack, 'BetaApiEndpoint', { value: betaApi.url! });
new CfnOutput(stack, 'Region', { value: Stack.of(stack).region});
