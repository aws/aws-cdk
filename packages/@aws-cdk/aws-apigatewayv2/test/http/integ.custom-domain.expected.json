{
  "Resources": {
    "echohandlerServiceRole833A8F7A": {
      "Type": "AWS::IAM::Role",
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com"
              }
            }
          ],
          "Version": "2012-10-17"
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition"
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
              ]
            ]
          }
        ]
      }
    },
    "echohandler8F648AB2": {
      "Type": "AWS::Lambda::Function",
      "Properties": {
        "Code": {
          "ZipFile": "exports.handler = async function(event, context) { return { statusCode: 200, headers: { \"content-type\": \"application/json\"  }, body: JSON.stringify(event) }; };"
        },
        "Handler": "index.handler",
        "Role": {
          "Fn::GetAtt": [
            "echohandlerServiceRole833A8F7A",
            "Arn"
          ]
        },
        "Runtime": "nodejs12.x"
      },
      "DependsOn": [
        "echohandlerServiceRole833A8F7A"
      ]
    },
    "DNFDC76583": {
      "Type": "AWS::ApiGatewayV2::DomainName",
      "Properties": {
        "DomainName": "apigv2.demo.com",
        "DomainNameConfigurations": [
          {
            "CertificateArn": "arn:aws:acm:us-east-1:111111111111:certificate",
            "EndpointType": "REGIONAL"
          }
        ]
      }
    },
    "HttpProxyProdApi368B6161": {
      "Type": "AWS::ApiGatewayV2::Api",
      "Properties": {
        "Name": "HttpProxyProdApi",
        "ProtocolType": "HTTP"
      }
    },
    "HttpProxyProdApiDefaultRouteinteghttpproxyHttpProxyProdApiDefaultRoute20082F68PermissionB29E0EB7": {
      "Type": "AWS::Lambda::Permission",
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "echohandler8F648AB2",
            "Arn"
          ]
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition"
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region"
              },
              ":",
              {
                "Ref": "AWS::AccountId"
              },
              ":",
              {
                "Ref": "HttpProxyProdApi368B6161"
              },
              "/*/*"
            ]
          ]
        }
      }
    },
    "HttpProxyProdApiDefaultRouteDefaultRouteIntegration702F0DF7": {
      "Type": "AWS::ApiGatewayV2::Integration",
      "Properties": {
        "ApiId": {
          "Ref": "HttpProxyProdApi368B6161"
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": {
          "Fn::GetAtt": [
            "echohandler8F648AB2",
            "Arn"
          ]
        },
        "PayloadFormatVersion": "2.0"
      }
    },
    "HttpProxyProdApiDefaultRoute40EFC108": {
      "Type": "AWS::ApiGatewayV2::Route",
      "Properties": {
        "ApiId": {
          "Ref": "HttpProxyProdApi368B6161"
        },
        "RouteKey": "$default",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "HttpProxyProdApiDefaultRouteDefaultRouteIntegration702F0DF7"
              }
            ]
          ]
        }
      }
    },
    "HttpProxyProdApiDefaultStage0038B180": {
      "Type": "AWS::ApiGatewayV2::Stage",
      "Properties": {
        "ApiId": {
          "Ref": "HttpProxyProdApi368B6161"
        },
        "StageName": "$default",
        "AutoDeploy": true
      },
      "DependsOn": [
        "DNFDC76583"
      ]
    },
    "HttpProxyProdApiDefaultStageinteghttpproxyDNfoo38514765": {
      "Type": "AWS::ApiGatewayV2::ApiMapping",
      "Properties": {
        "ApiId": {
          "Ref": "HttpProxyProdApi368B6161"
        },
        "DomainName": "apigv2.demo.com",
        "Stage": "$default",
        "ApiMappingKey": "foo"
      },
      "DependsOn": [
        "DNFDC76583",
        "HttpProxyProdApiDefaultStage0038B180"
      ]
    },
    "HttpProxyProdApiBetaStage7074CBDF": {
      "Type": "AWS::ApiGatewayV2::Stage",
      "Properties": {
        "ApiId": {
          "Ref": "HttpProxyProdApi368B6161"
        },
        "StageName": "beta",
        "AutoDeploy": true
      },
      "DependsOn": [
        "DNFDC76583"
      ]
    },
    "HttpProxyProdApiBetaStageinteghttpproxyDNbarA0E8C5B4": {
      "Type": "AWS::ApiGatewayV2::ApiMapping",
      "Properties": {
        "ApiId": {
          "Ref": "HttpProxyProdApi368B6161"
        },
        "DomainName": "apigv2.demo.com",
        "Stage": "beta",
        "ApiMappingKey": "bar"
      },
      "DependsOn": [
        "DNFDC76583",
        "HttpProxyProdApiBetaStage7074CBDF"
      ]
    },
    "DemoApiE67238F8": {
      "Type": "AWS::ApiGatewayV2::Api",
      "Properties": {
        "Name": "DemoApi",
        "ProtocolType": "HTTP"
      }
    },
    "DemoApiDefaultRouteinteghttpproxyDemoApiDefaultRoute050CFFE6PermissionB1FE82E7": {
      "Type": "AWS::Lambda::Permission",
      "Properties": {
        "Action": "lambda:InvokeFunction",
        "FunctionName": {
          "Fn::GetAtt": [
            "echohandler8F648AB2",
            "Arn"
          ]
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": {
          "Fn::Join": [
            "",
            [
              "arn:",
              {
                "Ref": "AWS::Partition"
              },
              ":execute-api:",
              {
                "Ref": "AWS::Region"
              },
              ":",
              {
                "Ref": "AWS::AccountId"
              },
              ":",
              {
                "Ref": "DemoApiE67238F8"
              },
              "/*/*"
            ]
          ]
        }
      }
    },
    "DemoApiDefaultRouteDefaultRouteIntegration714B9B03": {
      "Type": "AWS::ApiGatewayV2::Integration",
      "Properties": {
        "ApiId": {
          "Ref": "DemoApiE67238F8"
        },
        "IntegrationType": "AWS_PROXY",
        "IntegrationUri": {
          "Fn::GetAtt": [
            "echohandler8F648AB2",
            "Arn"
          ]
        },
        "PayloadFormatVersion": "2.0"
      }
    },
    "DemoApiDefaultRouteF8069A72": {
      "Type": "AWS::ApiGatewayV2::Route",
      "Properties": {
        "ApiId": {
          "Ref": "DemoApiE67238F8"
        },
        "RouteKey": "$default",
        "Target": {
          "Fn::Join": [
            "",
            [
              "integrations/",
              {
                "Ref": "DemoApiDefaultRouteDefaultRouteIntegration714B9B03"
              }
            ]
          ]
        }
      }
    },
    "DemoApiDefaultStage5FCE2787": {
      "Type": "AWS::ApiGatewayV2::Stage",
      "Properties": {
        "ApiId": {
          "Ref": "DemoApiE67238F8"
        },
        "StageName": "$default",
        "AutoDeploy": true
      },
      "DependsOn": [
        "DNFDC76583"
      ]
    },
    "DemoApiDefaultStageinteghttpproxyDNdemo045504F3": {
      "Type": "AWS::ApiGatewayV2::ApiMapping",
      "Properties": {
        "ApiId": {
          "Ref": "DemoApiE67238F8"
        },
        "DomainName": "apigv2.demo.com",
        "Stage": "$default",
        "ApiMappingKey": "demo"
      },
      "DependsOn": [
        "DemoApiDefaultStage5FCE2787",
        "DNFDC76583"
      ]
    }
  },
  "Outputs": {
    "RegionalDomainName": {
      "Value": {
        "Fn::GetAtt": [
          "DNFDC76583",
          "RegionalDomainName"
        ]
      }
    },
    "DomainName": {
      "Value": "apigv2.demo.com"
    },
    "CustomUDomainURL": {
      "Value": "https://apigv2.demo.com"
    },
    "ProdApiDefaultStageURL": {
      "Value": "https://apigv2.demo.com/foo"
    },
    "ProdApiBetaStageURL": {
      "Value": "https://apigv2.demo.com/bar"
    },
    "DemoApiDefaultStageURL": {
      "Value": "https://apigv2.demo.com/demo"
    }
  }
}